components/molecules/ContributorHoverCard/contributor-hover-card.tsx

```diff
@@ -1,66 +1,47 @@
-import { VscGitMerge, VscGitPullRequest } from "react-icons/vsc";
+import { useRouter } from "next/router";
 
 import Text from "components/atoms/Typography/text";
 import CardProfile from "../CardProfile/card-profile";
-import CardRepoList from "../CardRepoList/card-repo-list";
-import TestRepoAvatar from "public/icons/test-repo-avatar.svg";
+import CardRepoList, { RepoList } from "../CardRepoList/card-repo-list";
+import { calcDistanceFromToday } from "lib/utils/date-utils";
+import ContributorTable from "../ContributorTable/contributor-table";
 
-const ContributorHoverCard = () => {
-  const randomGithubHistory = Array.apply(null, Array(4));
-  const testRepoList = [
-    {
-      repoName: "test4",
-      repoIcon: TestRepoAvatar
-    },
-    {
-      repoName: "test5",
-      repoIcon: TestRepoAvatar
-    },
-    {
-      repoName: "test6",
-      repoIcon: TestRepoAvatar
-    },
-    {
-      repoName: "test6",
-      repoIcon: TestRepoAvatar
-    },
-    {
-      repoName: "test6",
-      repoIcon: TestRepoAvatar
-    },
-    {
-      repoName: "test6",
-      repoIcon: TestRepoAvatar
-    }
-  ];
+export interface ContributorsProfileType {
+  githubAvatar: string;
+  githubName: string;
+  totalPR: number;
+}
+interface ContributorHoverCardProps {
+  repoList: RepoList[];
+  githubAvatar: string;
+  githubName: string;
+  totalPR: number;
+  dateOfFirstPr: string;
+  topic?: string;
+}
+const ContributorHoverCard = ({
+  repoList,
+  githubName,
+  totalPR,
+  dateOfFirstPr,
+  githubAvatar
+}: ContributorHoverCardProps) => {
+  const router = useRouter();
+  const { filterName } = router.query;
+  const topic = filterName as string;
+
+  const calculatedDateFromToday = dateOfFirstPr ? calcDistanceFromToday(new Date(parseInt(dateOfFirstPr))) : "-";
   return (
-    <div className="w-[314px] gap-4 p-3 rounded-lg shadow-superlative flex flex-col">
+    <div className="w-[364px] bg-white gap-4 p-3 rounded-lg shadow-superlative flex flex-col">
       <div>
-        <CardProfile dateOfFirstPR="" githubName="Sungoldtech" totalPRs={45} />
-      </div>
-      <div className="bg-light-slate-3 text-xs px-2 py-1 text-light-slate-11 rounded">
-        <span>Latest PRs</span>
+        <CardProfile dateOfFirstPR={calculatedDateFromToday} githubName={githubName} totalPRs={totalPR} />
       </div>
-      <div className="flex flex-col gap-1.5">
-        {randomGithubHistory.map((item, index) => (
-          <div key={index} className="flex gap-1 text-xs justify-between items-center">
-            <div>
-              <VscGitPullRequest className="text-[14px]" color="green" />
-            </div>
-            <div>
-              <Text className="!text-light-slate-12">fix: increase paginated response lim </Text>
-            </div>
-            <div className="text-light-slate-11">2mo</div>
-          </div>
-        ))}
+      <div className="">
+        <ContributorTable limit={5} contributor={githubName} topic={topic} repositories={undefined} />
```

Comment:
> repositories={undefined}
Why is this undefined?

Reply:
It's optional and not required... I think a better approach is to not pass the props.
Thanks for pointing this out 👍